### PR TITLE
Add gocyclo to golangci-lint config

### PR DESCRIPTION
**Pull request by katexochen**
### Proposed change(s)

- Run gocyclo with golangci-lint, as the check is also used in the goreportcard

**Problems to solve:**
- [ ] access_manager/access_manager.go:109:1                                                    complexity 24 of func `deployKeys` is high (> 15)
- [ ] bootstrapper/cmd/bootstrapper/main.go:47:1                                                complexity 18 of func `main` is high (> 15)
- [ ] bootstrapper/internal/kubernetes/k8sapi/install.go:81:1                                   complexity 22 of func `(*osInstaller).extractArchive` is high (> 15)
- [ ] bootstrapper/internal/kubernetes/kubernetes.go:78:1                                       complexity 31 of func `(*KubeWrapper).InitCluster` is high (> 15)
- [ ] cli/internal/cmd/create.go:50:1                                                           complexity 17 of func `create` is high (> 15)
- [ ] hack/image-measurement/main.go:82:1                                                       complexity 16 of func `(*libvirtInstance).createLibvirtInstance` is high (> 15)
- [ ] hack/pseudo-version/pseudo-version.go:22:1                                                complexity 16 of func `main` is high (> 15)
- [ ] operators/constellation-node-operator/controllers/autoscalingstrategy_controller.go:45:1  complexity 29 of func `(*AutoscalingStrategyReconciler).Reconcile` is high (> 15)
- [ ] operators/constellation-node-operator/controllers/nodeimage_controller.go:79:1            complexity 22 of func `(*NodeImageReconciler).Reconcile` is high (> 15)
- [ ] operators/constellation-node-operator/controllers/nodeimage_controller.go:511:1           complexity 19 of func `(*NodeImageReconciler).createNewNodes` is high (> 15)
- [ ] operators/constellation-node-operator/controllers/nodeimage_controller.go:652:1           complexity 18 of func `nodeImageStatus` is high (> 15)
- [ ] operators/constellation-node-operator/controllers/pendingnode_controller.go:64:1          complexity 16 of func `(*PendingNodeReconciler).Reconcile` is high (> 15)
- [ ] operators/constellation-node-operator/internal/azure/client/nodeimage.go:50:1             complexity 16 of func `(*Client).CreateNode` is high (> 15)
- [ ] operators/constellation-node-operator/main.go:59:1                                        complexity 18 of func `main` is high (> 15)

**How does gocyclo compute complexity?**

> The cyclomatic complexity of a function is calculated according to the following rules:
> 
>  1 is the base complexity of a function
> +1 for each 'if', 'for', 'case', '&&' or '||'

### CODE REVIEW

**Comment by katexochen** [comment]
> Some of the functions look fine to me.

@Nirusu Which functions are you talking about exactly? There is also the possibility to disable the check for a function, as I did in the second commit for [NewFromDefault](https://github.com/edgelesssys/constellation/pull/185/files#diff-284021df5d4a42f236e823e18c539a12849770f88d6b1f5a3c06d111b1b90aadR69). But Imo the checker is right for most of the listed functions.

**Comment by thomasten** [comment]
The following is just my opinion. I think there's no right or wrong in this matter.

bootstrapper/cmd/bootstrapper/main.go:47:1 complexity 18 of func main
A dependency injection assembler function. You could split this, but I don't see a readability advantage.

bootstrapper/internal/kubernetes/k8sapi/install.go:81:1 complexity 22 of func (*osInstaller).extractArchive
This is one self-contained task. It would get harder to understand some details if split over multiple functions, but it wouldn't get any easier to understand the high level.

cli/internal/cmd/create.go:50:1 complexity 17 of func create
This looks super simple and clear. It's just UI. Logic is already separated. 

hack/image-measurement/main.go:82:1 complexity 16 of func (*libvirtInstance).createLibvirtInstance
Looks like an aggregate of similar actions.

I stopped here, but I think you can get the idea.

Bottom line: I tend to split functions if they involve very different actions or actions at different levels of abstraction. But it's ok and sometimes I prefer to have longer functions with similar kinds of actions instead of splitting them.

File: cli/internal/gcp/client/instances.go [hunk 1]
```diff
@@ -172,10 +179,12 @@ func (c *Client) TerminateInstances(ctx context.Context) error {
 		c.controlPlaneInstanceGroup = ""
 		c.controlPlanes = make(cloudtypes.Instances)
 	}
-	if err := c.waitForOperations(ctx, ops); err != nil {
-		return err
-	}
-	ops = []Operation{}
+
+	return c.waitForOperations(ctx, ops)
```
Comment (thomasten) [inline cli/internal/gcp/client/instances.go:183]: Won't this refactoring slow down the termination because it unnecessarily blocks here?

Comment (katexochen) [reply cli/internal/gcp/client/instances.go:183]: It waited before, still waits and it has to wait, because the templates can't be deleted as long as the instance group manager that uses it still exist.

Comment (thomasten) [reply cli/internal/gcp/client/instances.go:183]: I see. I didn't get the diff correctly and thought that there was only one wait before for all ops that now would be two waits.
